docs/fobnail-data-format.md

```diff
@@ -0,0 +1,103 @@
+# Fobnail data format
+
+Fobnail uses CBOR (with a few exceptions) for transmitting structured data both
+from Fobnail to Attester and from Attester to Fobnail. We use subset of
+[RFC7049](https://datatracker.ietf.org/doc/html/rfc7049) (no tagging and no
+extensions).
+
+## EK certificate
+
+EK certificate is transferred as a raw DER. We don't add any more data so there
+is no need to wrap it into CBOR.
+
+## AIK
+
+AIK is transferred in TPM format (TPM2B_PUBLIC structure). Re-encoding this into
+CBOR would strip essential data. Fobnail needs unmodified TPM2B_PUBLIC to
+successfully perform the "make credential" operation which is needed for
+credential activation.
+
+## Metadata
+
+Metadata is sent in CBOR-encoded format (also described
+[here](https://github.com/fobnail/docs/blob/main/dev-notes/metadata.md) in more
+detail). Example metadata looks like this (for readability, presented as JSON)
+
+```json
+{
+    // metadata version, used to detect backward-incompatible changes
+    // (major 0, unsigned integer)
+    "version": 1,
+    // device manufacturer extracted from SMBIOS tables (major 3, UTF-8 string)
+    "manufacturer": "Gigabyte",
+    // device model extracted from SMBIOS tables (major 3, UTF-8 string)
+    "model": "Gigabyte A520 AORUS ELITE",
+    // MAC address of primary network card (major 2, byte string)
+    "mac": [0xea, 0x96, 0x91, 0x87, 0x26, 0x8d],
+    // device serial number extracted from SMBIOS tables (major 3, UTF-8 string)
+    "sn": "S21N559B431",
+}
+```
+
+Metadata is signed by Attester using AIK before sending (see
+[Data signing](#data-signing) section below).
+
+## RIM
```

Comment:
I think we should redefine RIM format:

* similar but stripped (no `update_ctr` and `pcr` fields) format will be used for appraisal policy
* for appraisal policy order of hash algorithms matters, this is not possible to encode in current format
* there may appear additional hash algorithms for different TPMs

My suggestion is to switch from one map per hash (with map name = hash algorithm name) to array of maps (with algorithm name as key/value pair inside each map). We could also add a field specifying size of given hash, but I don't think it is necessary, it should be enough to test if all hashes in bank have the same length.

Reply:
@arturkow2000 is it feasible to do with current Fobnail code?